### ISSUE TITLE
List tables API gives one more (empty) page and terminates with empty string page_token instead of null

### ISSUE THREAD

**Issue by jaychia** [unitycatalog/unitycatalog#21]
When performing a tables.list operation, it seems that the server returns the empty string `""` instead of None as documented in the spec.

<img width="884" alt="image" src="https://github.com/unitycatalog/unitycatalog/assets/17691182/c7d88f8d-5081-48f4-bdaa-fd08d054b133">

On a default run of unity catalog:

```
!pip install unitycatalog

import unitycatalog

client = unitycatalog.Unitycatalog(base_url="http://127.0.0.1:8080/api/2.1/unity-catalog/")
response = client.tables.list(catalog_name="unity", schema_name="default")

# assert response.next_page_token is None, f"This will fail, with a non-empty token: {response.next_page_token}"

response = client.tables.list(catalog_name="unity", schema_name="default", page_token=response.next_page_token)
# assert len(response.tables) > 0, "This will fail, because response has no tables"

# assert response.next_page_token is None, f"This will fail, with empty token: {response.next_page_token}"
```

This led to an unexpected infinite loop on my client-side code! Expected behavior: return `null` for the token when the page stream is exhausted.

**Comment by abrassel** [comment]
Mind if I take this issue?

**Comment by tdas** [comment]
@abrassel oh yes! please take this up and make a PR! thank you so much in advance!

**Comment by abrassel** [comment]
thank you! @tdas can you assign it to me?

**Comment by abrassel** [comment]
@jaychia https://github.com/unitycatalog/unitycatalog/pull/70

**Comment by vikrantpuppala** [comment]
Fixed by #70
